cpp/src/arrow/compute/kernels/hash_aggregate_test.cc

```diff
@@ -2460,6 +2461,558 @@ TEST(GroupBy, Distinct) {
   }
 }
 
+MATCHER_P(AnyOfScalar, arrow_array, "") {
```

Comment:
IIRC, I think these convenience macros aren't always available in the CI environments we use. See https://github.com/apache/arrow/commit/cd30dea861d6dfd670032c655f329cb16bb99a7a